packages/wizard/src/lib/wizard.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useState } from 'react';
+
+import Tabs, { TabsProps as MUITabsProps } from '@mui/material/Tabs';
+import Box from '@mui/material/Box';
+
+import Tab from './Tab';
+import Header from './Header';
+import Footer from './Footer';
+import TabPanel from './TabPanel';
+import {
+  Step,
+  HeaderBaseProps,
+  FooterBaseProps,
+  ContentBaseProps,
+  TabBaseProps,
+} from './types';
+import { TabContainer } from './styles';
+import Content from './Content';
+
+function setAccessibilityProps(index: number) {
+  return {
+    id: `vertical-tab-${index}`,
+    'aria-controls': `vertical-tabpanel-${index}`,
+  };
+}
+
+interface StepStatus {
+  status: 'complete' | 'incomplete' | undefined;
+}
+
+export interface WizardProps {
+  onBack?: (stepIndex: number) => void;
+  onNext?: (stepIndex: number) => void;
+  onFinish?: () => void;
+  isLinear?: boolean;
+  showCompleted?: boolean;
+  TabOverride: React.FC<TabBaseProps>;
+  HeaderOverride: React.FC<HeaderBaseProps>;
+  ContentOverride: React.FC<ContentBaseProps>;
+  FooterOverride: React.FC<FooterBaseProps>;
+  steps: Step[];
+  TabsProps?: MUITabsProps;
+  tabIndicatorColor?: string;
+}
+
+const Wizard = ({
+  onBack,
+  onNext,
+  onFinish,
+  isLinear = false,
+  showCompleted = false,
+  tabIndicatorColor,
+  TabsProps,
+  TabOverride,
+  HeaderOverride,
+  ContentOverride,
+  FooterOverride,
+  steps,
+}: WizardProps) => {
+  const [currentStep, setCurrentStep] = useState(0);
+
+  const handleChange = (newStep: number) => {
+    const isNextStep = newStep === currentStep + 1;
+    const isDisabled = steps[newStep] ? steps[newStep].disabled : false;
+    if (isDisabled) return;
+    if (isLinear && !isNextStep) return;
+    setCurrentStep(newStep);
+  };
+
+  const handleFinish = () => {
+    onFinish && onFinish();
+  };
+
+  const handleNext = () => {
+    const isLastStep = currentStep >= steps.length - 1;
+
+    onNext && onNext(currentStep);
+    if (isLastStep) return;
+
+    handleChange(currentStep + 1);
+  };
+
+  const handleBack = () => {
+    const isFirstStep = currentStep === 0;
+
+    onBack && onBack(currentStep);
+    if (isFirstStep) return;
+
+    handleChange(currentStep - 1);
+  };
+
+  return (
+    <Box
+      sx={{
+        bgcolor: 'background.paper',
```

Comment:
we could provide a way to pass props to this container

Reply:
done!